src/sinks/vector/v2/sink.rs

```diff
@@ -28,10 +30,17 @@ pub struct VectorSink {
 impl VectorSink {
     async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
         input
-            .map(|mut event| EventData {
-                byte_size: event.size_of(),
-                finalizers: event.take_finalizers(),
-                wrapper: EventWrapper::from(event),
+            .map(|mut event| {
+                let byte_size = event.size_of();
+                emit!(&EventsReceived {
```

Comment:
I would have preferred doing this `emit` with batches of events to reduce the number of calls but the RFC says `immediately after creating or receiving one or more Vector events`.

Reply:
I'd prefer calling it from here https://github.com/vectordotdev/vector/blob/master/src/sinks/vector/v2/service.rs#L102

Reply:
Found the reason, dropping this commit as it's no longer needed 😉